scipy/interpolate/_bsplines.py

```diff
@@ -297,6 +298,72 @@ def basis_element(cls, t, extrapolate=True):
         c = np.zeros_like(t)
         c[k] = 1.
         return cls.construct_fast(t, c, k, extrapolate)
+    
+    @classmethod
+    def design_matrix(cls, x, t, k):
+        """
+        Returns a design matrix for BSpline object in "dense" or "CSR" formats.
+        
+        Parameters
+        ----------
+        x : array_like, shape (n,)
+            Points to evaluate the spline at.
+        t : array_like, shape (n,)
+            Sorted 1D array of knots.
```

Comment:
Probably want to discuss the constraints on the length of t given k. Also check and test eg length(t) == 2 and k=3 etc